platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/CheckedFingerprint.kt

```diff
@@ -27,8 +28,8 @@ sealed class CheckedFingerprint {
     object Valid : CheckedFingerprint()
 
     // The entry cannot be reused at all and should be recreated from scratch
-    class EntryInvalid(val reason: String) : CheckedFingerprint()
+    class EntryInvalid(val reason: StructuredMessage) : CheckedFingerprint()
 
     // The entry can be reused, however the values for certain projects cannot be reused and should be recreated
-    class ProjectsInvalid(val reason: String, val invalidProjects: Set<Path>) : CheckedFingerprint()
+    class ProjectsInvalid(val reason: StructuredMessage, val invalidProjects: Set<Path>) : CheckedFingerprint()
```

Comment:
Since CC report supports displaying structured messages for the invalidation reasons, we now craft them as such